src/simulators/statevector/qubitvector.hpp

```diff
@@ -1927,84 +1941,60 @@ std::vector<double> QubitVector<data_t>::probabilities(const reg_t &qubits) cons
   return probs;
 }
 
+template <typename data_t>
+void QubitVector<data_t>::get_accumulated_probabilities_vector(std::vector<double>& acc_probvector, 
+							       reg_t& index_vec) const {
+  uint_t size = 1LL << num_qubits_;
+  uint_t j = 1;
+  acc_probvector.push_back(0.0);
+  for (uint_t i=0; i<size; i++) {
+    if (!AER::Linalg::almost_equal(probability(i), 0.0)) {
+      index_vec.push_back(i);
+      acc_probvector.push_back(acc_probvector[j-1] + probability(i));
```

Comment:
`index_vec` and `acc_probvector` will require the same size of this qubit vector in worst case.

Reply:
In the worst case, you are correct. But in most cases, these vectors will be must smaller than the qubit vector.